pkg/resourcemanager/controller/garbagecollector/reconciler.go

```diff
@@ -70,7 +73,37 @@ func (r *Reconciler) Reconcile(reconcileCtx context.Context, _ reconcile.Request
 				continue
 			}
 
-			objectsToGarbageCollect[objectId{resource.kind, obj.Namespace, obj.Name}] = struct{}{}
+			if suppressString := obj.Annotations[managedresourcesutils.AnnotationKeySuppressGarbageCollectionUntil]; suppressString != "" {
+				suppressUntilTime, err := time.Parse(time.RFC3339, suppressString)
+				if err != nil {
+					log.Error(
+						err,
+						fmt.Sprintf( //nolint:logcheck // reason: Linter does not realise that message string is a constant
+							"Garbage collector: collective object has invalid '%s' annotation",
+							managedresourcesutils.AnnotationKeySuppressGarbageCollectionUntil),
+						"kind", obj.Kind,
+						"namespace", obj.Namespace,
+						"name", obj.Name,
+					)
+				} else {
+					// TODO: Andrey: P2: Update documentation to reflect maximum supported relative clock skew of 8
+					// minutes between the creator of a MR and any GC acting on that MR's secret.
```

Comment:
Why is this TODO not solved? IMO, it doesn't sound very useful to have this 8 minutes in place (where do the 8 minutes come from?).
In general, please enhance the documentation in `docs/concepts/resource-manager.md`.